src/System.CommandLine/Parsing/ParseOperation.cs

```diff
@@ -277,7 +285,12 @@ private void ParseDirectives()
         {
             while (More(out TokenType currentTokenType) && currentTokenType == TokenType.Directive)
             {
-                ParseDirective(); // kept in separate method to avoid JIT
+                if (_configuration.EnableDirectives)
```

Comment:
don't even try to parse the directives if the user has not asked for them (perf)